esphome/components/ams5915/ams5915.cpp

```diff
@@ -0,0 +1,181 @@
+#include "ams5915.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+namespace esphome {
+namespace ams5915 {
+static const char *const TAG = "ams5915";
+
+void Ams5915::set_transducer_type(Transducer model) { type_ = model; }
+
+/* starts the I2C communication and sets the pressure and temperature ranges using getTransducer */
+int Ams5915::begin_() {
+  // setting the min and max pressure based on the chip
+  this->get_transducer_();
+  // checking to see if we can talk with the sensor
+  for (size_t i = 0; i < max_attempts_; i++) {
+    status_ = read_bytes_(&pressure_counts_, &temperature_counts_);
+    if (status_ > 0) {
+      break;
+    }
+    delay(10);
+  }
+  return status_;
+}
+
+/* reads data from the sensor */
+int Ams5915::read_sensor_() {
+  // get pressure and temperature counts off transducer
+  this->status_ = read_bytes_(&this->pressure_counts_, &this->temperature_counts_);
+  // convert counts to pressure, PA
+  this->data_.pressure_pa_ =
+      (((float) (this->pressure_counts_ - this->dig_out_p_min_)) /
+           (((float) (this->dig_out_p_max_ - this->dig_out_p_min_)) / ((float) (this->p_max_ - this->p_min_))) +
+       (float) this->p_min_);
```

Comment:
I believe this is supposed to be remapping the value from the dig_out values to the chip based values? If so we have a `remap` function for this in `helpers.h`
```suggestion
  this->data_.pressure_pa_ = remap(this->pressure_counts_, DIGITAL_OUT_PRESSURE_MIN, DIGITAL_OUT_PRESSURE_MAX, this->pressure_min_, this->pressure_max_);
```